src/waveform/waveformwidgetfactory.cpp

```diff
@@ -688,8 +688,8 @@ void WaveformWidgetFactory::notifyZoomChange(WWaveformViewer* viewer) {
 }
 
 void WaveformWidgetFactory::renderSelf() {
-    ScopedTimer t("WaveformWidgetFactory::render() %1waveforms",
-            static_cast<int>(m_waveformWidgetHolders.size()));
+    ScopedTimer t("WaveformWidgetFactory::render() %1 waveforms",
+            QString::number(m_waveformWidgetHolders.size()));
```

Comment:
The idea was that all ScopeTimer statements have almost zero CPU overhead, when developer mode is disabled. 
QString does allocate memory even if the scoped timer is disabled so this is not the best idea. 

We may look for a solution that is equal fast in normal mode and faster in developer mode. This can be archived with the constexpr QLatin1String constructor. I have proposed here a Qt < 6.4 solution:   https://github.com/mixxxdj/mixxx/blob/751a104f4b21a2a74c1a9d0c1cd713ecd4605990/src/track/taglib/trackmetadata_file.cpp#L48-L54

Reply:
> The idea was that all ScopeTimer statements have almost zero CPU overhead, when developer mode is disabled.

Yes. I got that from the design and I tried to achieve the same here. This is the one case where the newer API does not exactly match the old one, but I figured this was a small-enough prize to pay in this case. There is also small-string-optimization, which should lessen the cost of the construction considerably. 

> We may look for a solution that is equal fast in normal mode and faster in developer mode. This can be archived with the constexpr QLatin1String constructor.

I'm afraid I can't quite follow you here. How does that help in the `QString::number` case?

Reply:
Just don't use QString::number. It is a heap allocating function. Does passing an int not work?

Reply:
No because the `QString::arg(Args&&...)` version only accepts objects implicitly convertible to int...

I can try creating a wrapper that does also accept numeric types and calls QString::number on them, though I don't know how complex the metaprogramming for that would be.